packages/studio-ui/src/utils/PageDataValidator.ts

```diff
@@ -28,12 +40,20 @@ export default class PageDataValidator {
    */
   validate(pageData: { pageName?: string; url?: string }): ValidationResult {
     const errorMessages: string[] = [];
-    if (pageData.pageName !== undefined)
-      errorMessages.push(...this.validatePageName(pageData.pageName));
-    if (pageData.url !== undefined)
-      errorMessages.push(
-        ...this.validateURLSlug(pageData.url, this.isEntityPage)
-      );
+    const { pageName, url } = pageData;
+    if (pageName !== undefined) {
+      errorMessages.push(...this.validatePageName(pageName));
+      if (this.pages[pageName]) {
+        errorMessages.push(`Page name "${pageName}" is already used.`);
+      }
+    }
+
+    if (this.isPagesJSRepo && !url) {
```

Comment:
Could we make a method called `validateURL` in this class? It would do this check and also invoke `this.validateURLSlug`. For code organization and readability purposes, I think having a consolidated `validateURL` method would be nice.